### PR TITLE
Add support for PostgreSQL regex match

### PR DESCRIPTION
**Pull request by b41sh**
Support PostgreSQL regular expression matching syntax as described in the Postgres manual: https://www.postgresql.org/docs/9.3/functions-matching.html

used by [arrow-datafusion#795](https://github.com/apache/arrow-datafusion/issues/795)

### CODE REVIEW

**Comment by alamb** [comment]
cc @Dandandan  / @andygrove

**Review by Dandandan — APPROVED** [review]

File: src/tokenizer.rs [hunk 1]
```diff
@@ -108,8 +108,14 @@ pub enum Token {
...
     Sharp,
-    /// Tilde `~` used for PostgreSQL Bitwise NOT operator
+    /// Tilde `~` used for PostgreSQL Bitwise NOT operator or case sensitive match regular operator
     Tilde,
+    /// `~*` , a case insensitive match regular operator in PostgreSQL
+    TildeAsterisk,
+    /// `!~` , a case sensitive not match regular operator in PostgreSQL
+    ExclamationMarkTilde,
+    /// `!~*` , a case insensitive not match regular operator in PostgreSQL
+    ExclamationMarkTildeAsterisk,
```
Comment (alamb) [inline src/tokenizer.rs:118]:
```suggestion
    /// `~*` , a case insensitive match regular expression operator in PostgreSQL
    TildeAsterisk,
    /// `!~` , a case sensitive not match regular expression operator in PostgreSQL
    ExclamationMarkTilde,
    /// `!~*` , a case insensitive not match regular expression operator in PostgreSQL
    ExclamationMarkTildeAsterisk,
```

File: src/tokenizer.rs [hunk 2]
```diff
@@ -1111,6 +1134,45 @@ mod tests {
...
+            Token::SingleQuotedString("^a".into()),
+            Token::Comma,
+            Token::Whitespace(Whitespace::Space),
+            Token::make_word("col", None),
+            Token::Whitespace(Whitespace::Space),
+            Token::ExclamationMarkTildeAsterisk,
+            Token::Whitespace(Whitespace::Space),
+            Token::SingleQuotedString("^a".into()),
+        ];
+        compare(expected, tokens);
```
Comment (alamb) [inline src/tokenizer.rs:1173]: 👍  this is quite cool

**Review by alamb — COMMENTED** [review]
This looks great @b41sh  -- FYI I pulled your code down locally to play around with it and tried to fix the clippy errors on CI while I was at it, in case that is helpful to you:  https://github.com/b41sh/sqlparser-rs/pull/1

**Comment by b41sh** [comment]
> This looks great @b41sh -- FYI I pulled your code down locally to play around with it and tried to fix the clippy errors on CI while I was at it, in case that is helpful to you: [b41sh#1](https://github.com/b41sh/sqlparser-rs/pull/1)

@alamb  Thanks for your help. But I have a question, why I run `cargo clippy` in my local environment doesn't report any errors or warnings?

**Comment by alamb** [comment]
> @alamb Thanks for your help. But I have a question, why I run cargo clippy in my local environment doesn't report any errors or warnings?

@b41sh  I don't honestly know / understand. I also had trouble getting clippy to report all errors locally for me until I switched to running it like this

```shell
find . -name '*.rs' | grep -v target | xargs touch && cargo clippy --all --all-targets -- -D warnings
```

While not ideal (it forces a recheck), it does work reliably for me

**Comment by b41sh** [comment]
> @b41sh  I don't honestly know / understand. I also had trouble getting clippy to report all errors locally for me until I switched to running it like this
> 
> ```shell
> find . -name '*.rs' | grep -v target | xargs touch && cargo clippy --all --all-targets -- -D warnings
> ```
> 
> While not ideal (it forces a recheck), it does work reliably for me

ok, I will try it

**Pull request merged**